File: frontend/src/types/career.ts
export interface ChatMessage {
  id: string;
  sender: 'user' | 'assistant';
  text: string;
  timestamp: string;
  suggested_actions?: string[];
}

export interface LearningRecommendation {
  id: string;
  employee_id: string;
  skill_gap_id?: string | null;
  title: string;
  description: string;
  resource_type: string;
  priority: string;
}

File: backend/app/schemas/career.py
from typing import List, Optional
from pydantic import BaseModel, Field


class CareerChatRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=2000)


class CareerChatResponse(BaseModel):
    response: str
    suggested_actions: List[str] = []


class LearningRecommendationRead(BaseModel):
    id: str
    employee_id: str
    skill_gap_id: Optional[str] = None
    title: str
    description: str
    resource_type: str
    priority: str

    model_config = {"from_attributes": True}
